feat(prettier): fill printing `CallExpression` implementation

Dunqing opened the pull request:
Printing a call expression correctly is mostly done, except for printing comments. Not many tests have passed because there is a problem with `ParentStack`. Some places use `ParentStack` to check the current parent, which may be incorrect. The issue is that some printing implementations do not call `push`/`pop`, meaning the recorded parent node is not accurate—it only reflects the last pushed record. I’ve attempted to solve this problem in #11339.

Dunqing commented on crates/oxc_formatter/src/write/call_arguments.rs:
```diff
@@ -0,0 +1,1271 @@
+                        Argument::FunctionExpression(_) | Argument::ArrowFunctionExpression(_)
+                    )
+                });
+            }
```
Good catch!

Boshen commented:
### Merge activity

* **May 30, 2:32 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **May 30, 2:32 PM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Dunqing committed:
feat(prettier): fill printing `CallExpression` implementation (#11299)

Printing a call expression correctly is mostly done, except for printing comments. Not many tests have passed because there is a problem with `ParentStack`. Some places use `ParentStack` to check the current parent, which may be incorrect. The issue is that some printing implementations do not call `push`/`pop`, meaning the recorded parent node is not accurate—it only reflects the last pushed record. I’ve attempted to solve this problem in #11339.
